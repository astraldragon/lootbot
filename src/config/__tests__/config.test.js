jest.mock("../../logger", () => ({
  info: jest.fn("info")
}));

const logger = require("../../logger");
const config = require("../config");

describe("config/config", () => {
  it("should have the proper database config", () => {
    expect(config).toMatchSnapshot();
  });

  it("should configure logging for development config", () => {
    const msg = "my message";
    config.development.logging(msg);
    expect(logger.info).toHaveBeenCalledWith(msg);
  });

  it("should configure logging for production config", () => {
    const msg = "my message";
    config.production.logging(msg);
    expect(logger.info).toHaveBeenCalledWith(msg);
  });
});
