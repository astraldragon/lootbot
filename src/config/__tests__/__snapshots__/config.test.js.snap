// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config/config should have the proper database config 1`] = `
Object {
  "development": Object {
    "dialect": "sqlite",
    "operatorsAliases": false,
    "seederStorage": "sequelize",
    "storage": "lootbot.sqlite",
  },
  "production": Object {
    "dialect": "postgres",
    "operatorsAliases": false,
    "seederStorage": "sequelize",
    "use_env_variable": "DATABASE_URL",
  },
}
`;
